### PR TITLE
feat: hook typings

### LINKED ISSUE #421: Hook types incomplete
### Version
@nuxt/content: 1.7.0
nuxt: 2.14.1	

### Steps to reproduce
When using hooks in the nuxt config TypeScript warns me for incomplete types of the hook builder by Nuxt.

```
//nuxt.config.ts
export default {
  modules: [,
    '@nuxt/content'
  ],
  hooks: {
    'content:file:beforeInsert': (document) => {
      if (document.extension === '.md') {
        const { time } = require('reading-time')(document.text)

        document.readingTime = time
      }
    }
  }
}
```

TypeScript doesn't know the `content:file:beforeInsert` function (and other hooks)

### What is Expected?
That TypeScript and the Nuxt type config knows these options.

### What is actually happening?
The Nuxt content decalrations should be expanded for hooks: 
https://github.com/nuxt/content/blob/1dff4e9e9c4bcff03396d711c37f5eee70b8690c/packages/content/types/index.d.ts#L35

As a start I tried the following:

```

declare module "@nuxt/types" {
	...
	interface Configuration {
		"content:file:beforeInsert": (document: any) => void;
	}
	...
}

### PR DESCRIPTION
**Pull request by mathe42**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

fixes #421 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by mathe42** [comment]
I will wait for #442 to be merged first to add types for that PR.

**Comment by Atinux** [comment]
Nice work @mathe42 :)

File: packages/content/types/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import '@nuxt/types'
+import '@nuxt/types/config/hooks'
```
Comment (dipsaus9) [inline packages/content/types/index.d.ts:2]: I don't believe this import is necessary as the `@nuxt/types` also imports the config/hooks. In my setup it wasn't necessary, can be mistaken

Comment (mathe42) [reply packages/content/types/index.d.ts:2]: You might be right I didn't test this - so I wantet to be sure ;D

**Pull request closed without merging**